ReelTalk/index.ios.js

```diff
@@ -1,87 +1,271 @@
 'use strict';
 
-var React = require('react-native');
-var {
+import React, {
   AppRegistry,
-  NavigatorIOS,
   StyleSheet,
   TabBarIOS,
-} = React;
+  Text,
+  Navigator,
+  NavigatorIOS,
+  View,
+  TouchableOpacity,
+  TouchableHighlight
+} from 'react-native';
 
-var RecommendScreen = require('./screens/RecommendScreen');
-var ListsScreen = require('./screens/ListsScreen');
-var TopChartsScreen = require('./screens/TopChartsScreen');
-var SettingsScreen = require('./screens/SettingsScreen');
+import RecommendScreen from './screens/RecommendScreen';
+import ListsScreen from './screens/ListsScreen';
+import TopChartsScreen from './screens/TopChartsScreen';
+import SettingsScreen from './screens/SettingsScreen';
 
-var ReelTalk = React.createClass({
+import cssVar from 'cssVar';
+
+const NavigationBarRouteMapper = {
+  LeftButton: function(route, navigator, index, navState) {
+    if (index === 0) {
+      return null;
+    }
+
+    const previousRoute = navState.routeStack[index - 1];
+    return (
+      <TouchableOpacity
+        onPress={() => navigator.pop()}
+        style={styles.navBarLeftButton}>
+        <Text style={[styles.navBarText, styles.navBarButtonText]}>
+          {"< " + previousRoute.title}
+        </Text>
+      </TouchableOpacity>
+    );
+  },
+
+  RightButton: function(route, navigator, index, navState) {
+    return null;
+  },
+
+  Title: function(route, navigator, index, navState) {
+    return (
+      <Text style={[styles.navBarText, styles.navBarTitleText]}>
+        {route.title}
+      </Text>
+    );
+  },
+
+};
+
+const Main = React.createClass({
+  getInitialState: function() {
+    return { state: this.props.activeTab }
+  },
+
+  componentDidMount: function() {
+    this.refs.recommendTabRef.setState({hasBeenSelected: true})
+  },
+
+  renderRecommendScreen: function() {
```

Comment:
I moved it because `Main` is using iOS specific components: `Tabbar` amd `TabBar.Item`.

We can move this into a file called `MainIOS.js` but it won't be something reusable (tabs and nested navigators are always at the app's route level) so I did not see the point beyond line count.

The Android implementation uses the `screens` too, for now. We can either inject components into them or (i think better) built iOS and Android specific screens when needed, composing the universal components like Lolomo.